linera-views/src/dynamo_db.rs

```diff
@@ -69,10 +70,21 @@ const VALUE_ATTRIBUTE: &str = "item_value";
 /// The attribute for obtaining the primary key (used as a sort key) with the stored value.
 const KEY_VALUE_ATTRIBUTE: &str = "item_key, item_value";
 
-/// TODO(#1084): We remove 600 from 400 * 1024. We need to determine what exactly needs to be done.
+/// TODO(#1084): The scheme below with the MAX_VALUE_BYTES has to be checked
+/// This is the maximum size of a raw value in DynamoDb.
+const RAW_MAX_VALUE_BYTES: usize = 409600;
+
 /// Fundamental constants in DynamoDB: The maximum size of a value is 400KB
 /// See https://docs.aws.amazon.com/amazondynamodb/latest/developerguide/ServiceQuotas.html
-const MAX_VALUE_BYTES: usize = 409000;
+/// However, the value being written can also be the serialization of a SimpleUnorderedBatch
+/// Therefore the actual MAX_VALUE_BYTES might be lower.
+/// At the maximum the key_size is 1024 bytes (see below) and we pack just one entry.
+/// So if the key has 1024 bytes this gets us the inequality
+/// 1 + 1 + serialized_size(1024)? + serialized_size(x)? <= 400*1024
+/// and so this simplifies to 1 + 1 + (2 + 1024) + (3 + x) <= 400 * 1024
+/// (we write 3 because get_bitsize(400*1024) = 3)
+/// and so to a maximal value of 408569;
```

Comment:
So this should be `RAW_MAX_VALUE_BYTES - 1024 - serialized_size(&SimpleUnorderedBatch::default())`? Would be good to have a test that asserts that.

Reply:
It is more complicated than that because the `get_bitsize` contribute to the offset, but I will work on some test.